### ISSUE TITLE
Create UI for a "detect bugs with similar descriptions" feature

### ISSUE THREAD

**Issue by MDTsai** [webcompat/webcompat.com#1846]
It's from https://github.com/webcompat/webcompat.com/issues/1826#issuecomment-333793412

**Intro:**
When there is Firefox regression bugs, we'll see users report same issues but they have same root cause. It comes into my mind if we can let user know other users already reported similar issue, they can see what's happening and vote/+1/vouch same issue happen to him/her.

**Description:**
While user is typing in the description and more than 2 words, use the keyword to search existing issue description and find similar description.

![2017-10-20 11-49-49](https://user-images.githubusercontent.com/5763237/31804884-2ced5468-b58f-11e7-865c-c2368969710d.png)

Click on the suggested issue opens a new tab, see what's going on and do the following. Agree with +1 or disagree then close the tab.

**TBD:**
- What's the better way to guide users have discussion similar description in an issue?
- It's doable via github search API or we need another DB/end point to do this?

**Comment by zoepage** [comment]
@MDTsai 
So as far as I understand this, it's very close to the initial proposal in https://github.com/webcompat/webcompat.com/issues/1826#issue-258749417, just not filtering by domain, but keywords in description. Correct?
Would this be included in the bug report form or would this be a separate tool?

> It's doable via github search API or we need another DB/end point to do this?

It is doable, but gives back very different results depending on keywords. e.g. when you look for a domain firefox.org, firefox org and https://firefox.org would give you different results. If we are okay with this, we could use the github search API and go from there. (see more discussion here https://github.com/webcompat/webcompat.com/issues/1826#issuecomment-331373878)

> What's the better way to guide users have discussion similar description in an issue?

That is a very good question. I think, the answer depends on who we want to provide the feature for. 
In one of the discussions we had a few months ago, we talked about 2 different reporter types. The ones, who want to report really quickly an issue and move on (user, occasionally reporting bugs) and the power reporter, who would love to have as many features as possible while reporting (e.g. sprints, Q&A sessions). I feel like, we move more and more towards 2 versions of the bug report form. One that is very basic for non-logged in users and one which feels more and more like Bugzilla for logged in users. Maybe we should also define a long term goal to achieve for the bug form? (Sorry, for being so meta about this question. This feature is a great idea and we should proceed with it.)

**Comment by karlcow** [comment]
As @zoepage said above, this is similar than the discussion in #1826 because the mechanics is the same. I will copy/paste my last comment there:

> Maybe the first prototype, not something that would ever go to the production site but could be deployed on staging. Would be to just try to pull the search issues list when typing the domain and see if we get something meaningful. Not triggering any further action. We might not have to even do it on the site at first stage.
> 
>     simple web page, showing the search results when typing in an input field. (somewhere on the web)
>     Simple search list on the form (on staging)
>     Next steps and so on.
> 
> Like this we don't develop a big machine, we learn from what we see, we try to tweak if we can improve the data quality, we learn what kind of actions we can do with this list or abandon if not fruitful.
> 
> That makes the goal less ambitious, but maybe more achievable through small iterations.

Let's learn what we get from a simple page experimenting ideas. And then if we think we get something meaningful enough, we could push to the next step. Aka not putting the cart before the horses or the plow before the cow  🐄 🤡

**Comment by zoepage** [comment]
@MDTsai You had the idea to also integrate bugzilla issues info this as far as I remember?

**Comment by MDTsai** [comment]
@zoepage  Not sure what you mean "integrate bugzilla issues info"? I did a [dependency graph prototype](https://mdtsai.github.io/wc-bz.html) which helps us to track the dependency between issues and bugs, to see if issues have same root cause. Now I'm doing is to improve the prototype to become more responsive, support to query with an interval (like dependency last week).

Agree with the long term direction, we have a lot ideas to improve each part of bug quality and other things, but that may conduct different experience when file bug.

**Comment by zoepage** [comment]
@MDTsai I remember you mentioning, that you wanted to add a way to find duplicates not just in our issue repository, but also in the bugzilla issues. So we could mark both. Does this make sense?

**Comment by MDTsai** [comment]
Yes, mark both is necessary, I did this via filter comments with URL to webcompat issue and bugzilla bug URL.

**Comment by zoepage** [comment]
I think, this would be a huge improvement! :)

**Comment by zoepage** [comment]
@MDTsai and I talked about this feature and decided to collaborate on it. 

#### First step (Q4/17):
- Build prototype 
   - filter by URL 
   - use GitHub search API
   - either use git sub-modules or standalone 

We dismiss performance as well as design here and take care of those issues later in the process.

*Goal: Good for internal testing phase*

#### Second step (Q1/18):
- Re-evaluating prototype.
- Moving issues to local database (we need to decide here on what to use to store / filter issues).
- Adding a way to add Bugzilla link to a duplicate issue.

*Goal: Good for regular internal usage.*

#### Third step (Q2 / Q3):
- Testing phase + UX Audit
- implementing into webcompat.com (bug form || standalone feature)

*Goal: Roll out to all users.*

**Comment by zoepage** [comment]
@MDTsai and I talked about this feature and decided to collaborate on it. 

#### First step (Q4/17):
- Build prototype 
   - filter by URL 
   - use GitHub search API
   - either use git sub-modules or standalone 

We dismiss performance as well as design here and take care of those issues later in the process.

*Goal: Good for internal testing phase*

#### Second step (Q1/18):
- Re-evaluating prototype.
- Moving issues to local database (we need to decide here on what to use to store / filter issues).
- Adding a way to add Bugzilla link to a duplicate issue.

*Goal: Good for regular internal usage.*

#### Third step (Q2 / Q3):
- Testing phase + UX Audit
- implementing into webcompat.com (bug form || standalone feature)

*Goal: Roll out to all users.*

**Comment by MDTsai** [comment]
Created one: https://github.com/webcompat/webcompat.com/tree/similarbugs

**TODOs:**
- [ ] UI works
- [ ] Connect with [github search API](https://developer.github.com/v3/search/)
- [ ] Test plan
